### PR TITLE
feat(theme-doc): add footer

### PR DESCRIPTION
**Pull request by cijiugechu**
PC: 
PC
mobile:
mobile

### CODE REVIEW

File: packages/theme-doc/src/Layout/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import React, { useContext, useMemo } from 'react'
+import RcFooter from 'rc-footer'
+
+import { themeConfigCtx, useThemeCtx } from '../ctx'
+import s from './Footer.module.less'
+
+export type FooterLink = {
```
Comment (csr632) [inline packages/theme-doc/src/Layout/Footer.tsx:7]: Could we allow users to pass options `icon`, `description`, `openExternal` of [rc-footer](https://www.npmjs.com/package/rc-footer)? I think they are useful to users.

image

Example usage: antd document:

https://github.com/ant-design/ant-design/blob/01087b2571ee26629b0df78e5a76c788572cebdb/site/theme/template/Layout/Footer.tsx#L46

Outcome:
image

**Comment by csr632** [comment]
Nice feature! Thanks!
I made some changes to style so that we can reuse style code from rc-footer.

File: packages/theme-doc/src/Layout/Footer.module.less [hunk 2]
```diff
@@ -0,0 +1,30 @@
+@import 'rc-footer/assets/index.less';
+
+@footer-prefix-cls: footer;
```
Comment (csr632) [inline packages/theme-doc/src/Layout/Footer.module.less:3]: We can import a less file and overwrite variables in it like this.
Ref: https://stackoverflow.com/a/19514889

**Comment by csr632** [comment]
> Is there a need to add FooterConfig to doc-site like use-theme-doc's config ? Maybe this will help users get intuitive experience .

I just added it. 

image

The dark theme of `rc-footer` looks a bit strange in theme-doc. Maybe we should try [light theme of rc-footer](https://www.npmjs.com/package/rc-footer)...

**Comment by csr632** [comment]
Just added support for footer theme. You can take a look and change it :)

**Comment by cijiugechu** [comment]
> Just added support for footer theme. You can take a look. Feel free to improve it :)
> 
> Light theme: image
> 
> Dark theme: image

Thanks for your improvements! Just my opinion, the footer contains external or additional informations that distinct from main contents , so probably adding some visual distinctions such as backgroundColor to them will be better.
Just like: 
屏幕截图(24)

**Pull request merged**